### PR TITLE
Addressbooklist in a popover

### PR DESCRIPTION
**Pull request by skjnldsv**
Testing: 
[contacts.tar.gz](https://github.com/nextcloud/contacts/files/749683/contacts.tar.gz)

- [x] Popover structure
    - [x] Css guidelines
    - [x] Don't close settings after click
- [x] Clipboard
    - [x] Tooltips
    - [x] Fallback to input if no clipboard support
- [x] Shares design 
    - [x] List
    - [x] Checkboxes
    - [x] Don't close settings after click

Fix and finishes #14
capture d ecran_2017-02-18_18-14-11 capture d ecran_2017-02-18_18-15-23 capture d ecran_2017-02-18_18-15-47

### CODE REVIEW

**Comment by jancborchardt** [comment]
Very nice work! Just 3 details which are probably related to the core styles:
- top right border radius is not set
- there is a bit too much bottom padding on the popover
- »Share« and »Delete« are much lighter than the other item. Were you hovering »Download«? Cause they should be the same lightness.

And one thing: Because sharing is inside the menu, »Copy link« should be too. Because otherwise the icon without text will cause confusion because it’s used in the context of sharing usually. Yes, for sharing a link, but mostly in combination with the other sharing. :)

**Comment by skjnldsv** [comment]
@jancborchardt Thanks! :)
- I wasn't hovering it, there is some fix to do. 
- Top right indeed comes form core, will fix.
- Makes sense for the copy link/share. Should we put the download button outside then?

**Comment by skjnldsv** [comment]
This is what it does on stable 11 @jancborchardt 
capture d ecran_2017-01-25_15-55-44

**Comment by skjnldsv** [comment]
Backport on nextcloud/server#3286

**Comment by MorrisJobke** [comment]
For me those icons don't appear in a popover but inline in the row. 😕 So the three dots menu is not there.

**Comment by skjnldsv** [comment]
@MorrisJobke I added a contact build for testing. :)
But we need this backport! https://github.com/nextcloud/server/pull/3286

**Comment by skjnldsv** [comment]
Rebased.

We now should wait for 11.0.2

**Comment by skjnldsv** [comment]
All clear, rebased and fixed! (see updated screenshots on first post)
Please review @nextcloud/contacts :)

**Comment by jancborchardt** [comment]
@skjnldsv can you post a screenshot of how it looks like now? Am at a fair the week and can't test well :/

How it should be: Sharing icon outside of the popover, but the Copy link function inside. And it should ideally trigger a copy directly on clicking the entry.

**Comment by skjnldsv** [comment]
@jancborchardt The first post screenshots are up-to-date :)

You want me to move the share url into the popover? Also we could completely get rid of the text input for the link and simply copy the link by clicking the entry IN the popovermenu :)

**Comment by jancborchardt** [comment]
@skjnldsv yes, exactly what you said :) move the "Copy sync link" to the menu and make it copy on click. You can use tooltip.js to show a "Link copied" confirmation.

**Comment by skjnldsv** [comment]
Can't get the clipboard to work, don't know why :/
Can somebody help me there? Is it working for you?

I added some feedback on the console as a debug, but even if everything seems normal, my clipboard stays empty 😢

File: js/components/addressBook/addressBook_controller.js [hunk 1]
```diff
@@ -141,6 +141,12 @@ angular.module('contactsApp')
 	};
 
 	ctrl.onSelectSharee = function (item) {
+		// Prevent settings to slide down
+		$('#app-settings-header > button').data('apps-slide-toggle', false);
```
Comment (skjnldsv) [inline js/components/addressBook/addressBook_controller.js]: This is an ugly trick. But core is broken for popups in the settings area.
Here is the issue: https://github.com/nextcloud/server/blob/25acce7e0d992b10b930472b8138240b7bd9af13/core/js/apps.js#L108

**Comment by skjnldsv** [comment]
@nextcloud/designers @nextcloud/contacts All clear.
Screenshots are updated on first post.

Please review! 🐤

**Comment by irgendwie** [comment]
@skjnldsv The 'Copy URL' tooltip is a bit glitchy for me. When clicking 'Copy URL' the 'Copied!' tooltip appears shortly in the top left corner of the screen. When hovering the 'Copy URL' (after closing and reopening the settings) the tooltip still shows 'Copied!' - only after some time it changes back to the original help text.

**Comment by jancborchardt** [comment]
@skjnldsv any update? Would be great if we can get this in time for Nextcloud 12 :)

@irgendwie I would say the tooltip glitch we can fix separately? :) Let’s move forward and not block this PR too long.

**Comment by skjnldsv** [comment]
@jancborchardt this update is ready to be merged.
Just the tooltip bug @irgendwie had.

**Comment by irgendwie** [comment]
@jancborchardt Would love to merge, still it would break the contacts app for nc 9 and nc 10 and would only be compatible with nc 11 and 12. Are we willing to do this?

**Comment by skjnldsv** [comment]
For the record, I am :)

**Comment by jancborchardt** [comment]
cc @LukasReschke @MorrisJobke @jospoortvliet for second opinions.

I would agree with @skjnldsv it’s ok for the Contacts app to only support the two latest releases. It’s a big testing workload otherwise and also contacts syncing is in core anyway. And the app in its current state is still available for Nextcloud 9 and 10.
We want people to update to new versions also for security reasons, so getting proper updated apps is another push to do so.

**Comment by Henni** [comment]
@jancborchardt I agree. But we should really support the latest two versions and thus might want to wait for the nc 12 release.

**Comment by jancborchardt** [comment]
> @jancborchardt I agree. But we should really support the latest two versions and thus might want to wait for the nc 12 release.

Yes, good call – do you mean after or before the Nextcloud 12 release? And for when is the next release of the Contacts app planned? Nextcloud 12 will come soon. :)

**Comment by Henni** [comment]
@jancborchardt how about the same day as Nextcloud 12? :wink: 

The current milestone has a due date of April 1st. But it might make sense to delay it to the nc release.

**Comment by jancborchardt** [comment]
@Henni cool, let’s do that! :) But then let’s also merge this change already so it can be used and tested as early as possible. :) Ok @irgendwie @Henni @skjnldsv?

**Pull request merged**